packages/logs/lib/otlp/otlpSpan.ts

```diff
@@ -0,0 +1,109 @@
+import type { MessageRow, MessageState, OperationRow } from '@nangohq/types';
+import { SpanStatusCode } from '@opentelemetry/api';
+import type { Span } from '@opentelemetry/api';
+import { otlp, otlpRoutingAttributeKey } from './otlp.js';
+import { envs } from '../env.js';
+
+export class OtlpSpan {
+    private span: Span | null = null;
+
+    constructor(operation: OperationRow) {
+        if (!envs.NANGO_LOGS_ENABLED || !shouldTrace(operation)) {
+            return;
+        }
+        // if no environmentId, we cannot route the span to the correct exporter so we don't start it
+        if (operation.environmentId) {
+            const attributes: Record<string, any> = {
+                [otlpRoutingAttributeKey]: `environment:${operation.environmentId}`,
+                'nango.operation.id': operation.id,
+                'nango.operation.type': operation.operation.type,
+                'nango.operation.action': operation.operation.action,
+                'nango.operation.message': operation.message,
+                'nango.account': operation.accountName
+            };
+            if (operation.environmentName) {
+                attributes['nango.environment'] = operation.environmentName;
+            }
+            if (operation.providerName) {
+                attributes['nango.provider'] = operation.providerName;
+            }
+            if (operation.integrationName) {
+                attributes['nango.integration'] = operation.integrationName;
+            }
+            if (operation.connectionName) {
+                attributes['nango.connection'] = operation.connectionName;
+            }
+            if (operation.syncConfigName) {
+                attributes['nango.sync'] = operation.syncConfigName;
+            }
+
+            const spanName = `nango.${operation.operation.type}.${operation.operation.action}`.toLowerCase();
+            this.span = otlp.tracer.startSpan(spanName, { attributes });
+        }
+    }
+
+    fail(err: Error): void {
+        if (this.span) {
+            this.span.recordException(err);
+            this.span.setStatus({ code: SpanStatusCode.ERROR });
+        }
+    }
+
+    end(state: MessageState): void {
+        if (this.span) {
+            this.span.setAttribute('nango.operation.status', state);
+            this.span.setStatus({ code: ['success', 'waiting', 'running'].includes(state) ? SpanStatusCode.OK : SpanStatusCode.ERROR });
+            this.span.end();
+        }
+    }
+
+    enrich(data: Partial<MessageRow>): void {
```

Comment:
We need to make sure `enrich` always comes before send, in #2884 I fixed a bunch of those but might be worth checking again

Reply:
yes I expect to tweak a few things based on the first feedback. Even the subset of attributes attaches to traces is arbitrary I might have missed some important info but it will be easier to add later on